### PR TITLE
Allow connection argument in env.expect()

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,14 +34,14 @@ jobs:
...
 
-    - name: Cache pip
-      uses: actions/cache@v1
-      with:
-        path: ~/.cache/pip # This path is specific to Ubuntu
-        # Look to see if there is a cache hit for the corresponding requirements file
-        key: ${{ matrix.platform }}-${{ matrix.python }}-pyproject.toml-${{ hashFiles('pyproject.toml') }}
-        restore-keys: |
-          ${{ matrix.platform }}-${{ matrix.python }}-pyproject.toml-${{hashFiles('pyproject.toml')}}}
+#    - name: Cache pip
```
Comment (MeirShpilraien) [inline .github/workflows/ci.yml:37]: Why disable it?

File: RLTest/env.py [hunk 2]
```diff
@@ -28,15 +28,19 @@ def method(*argc, **nargs):
 
 
 class Query:
-    def __init__(self, env, *query):
+    def __init__(self, env, *query, **kwargs):
```
Comment (MeirShpilraien) [inline RLTest/env.py:31]: Can you explain why you needed this feature?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rafie) [reply .github/workflows/ci.yml:37]: Only diagnostic, will revert that.

File: RLTest/env.py (hunk 2 above)
Comment (rafie) [reply RLTest/env.py:31]: To allow something like: `env.expect(..., conn: r).equal(...)` when `r` is a cluster connection.

Comment (rafie) [reply RLTest/env.py:31]: To allow `env.expect(..., conn: r).equal(...)` with `r` as a cluster connection.

File: RLTest/env.py [hunk 3]
```diff
@@ -28,14 +28,18 @@ def method(*argc, **nargs):
...
 
 class Query:
-    def __init__(self, env, *query, **options):
+    def __init__(self, env, *query, **kwargs):
         self.query = query
         self.options = options
         self.env = env
+        self.conn = kwargs.pop('conn', None)
+        if self.conn is None:
+            self.conn = env.con
```
Comment (oshadmi) [inline RLTest/env.py:37]: Can be shortened?
```suggestion
        self.conn = kwargs.pop('conn', env.con)
```

File: RLTest/env.py [hunk 4]
```diff
@@ -433,11 +437,15 @@ def assertIsInstance(self, value, instance, depth=0):
...
+    def expect(self, *query, **kwargs):
+        conn = kwargs.pop('conn', None)
+        return Query(self, conn=conn, *query)
 
-    def cmd(self, *query, **options):
-        res = self.con.execute_command(*query, **options)
+    def cmd(self, *query, **kwargs):
+        conn = kwargs.pop('conn', None)
+        if conn is None:
+            conn = self.con
```
Comment (oshadmi) [inline RLTest/env.py:447]: Can be shortened?

File: RLTest/exists_redis.py [hunk 5]
```diff
@@ -16,6 +17,7 @@ def __init__(self, addr='localhost:6379', password = None, **kwargs):
         self.host, self.port = addr.split(':')
         self.port = int(self.port)
         self.password = password
+        self.useTLS = kwargs['useTLS']
```
Comment (oshadmi) [inline RLTest/exists_redis.py:20]: Exists always? (no need to use `kwargs.get` with a default `None`?)

**Pull request merged**